pkg/e2e/infra/utils.go

```diff
@@ -155,6 +236,40 @@ func deleteMachine(client runtimeclient.Client, machine *mapiv1beta1.Machine) er
 	})
 }
 
+// getNodesFromMachineSet returns an array of nodes backed by machines owned by a given machineSet
+func getNodesFromMachineSet(client runtimeclient.Client, machineSet mapiv1beta1.MachineSet) ([]*corev1.Node, error) {
+	machines, err := getMachinesFromMachineSet(client, machineSet)
+	if err != nil {
+		return nil, fmt.Errorf("error calling getMachinesFromMachineSet %v", err)
+	}
+
+	var nodes []*corev1.Node
+	for key := range machines {
+		node, err := getNodeFromMachine(client, machines[key])
+		if err != nil {
+			return nil, fmt.Errorf("error getting node from machine %q: %v", machines[key].Name, err)
+		}
+		nodes = append(nodes, node)
+	}
+	return nodes, nil
+}
+
+// getNodeFromMachine returns the node object referenced by machine.Status.NodeRef
+func getNodeFromMachine(client runtimeclient.Client, machine mapiv1beta1.Machine) (*corev1.Node, error) {
+	var node corev1.Node
+	if machine.Status.NodeRef == nil {
+		glog.Infof("machine %q has no NodeRef", machine.Name)
+		return nil, nil
+	}
+	key := runtimeclient.ObjectKey{Namespace: machine.Status.NodeRef.Namespace, Name: machine.Status.NodeRef.Name}
+	if err := client.Get(context.Background(), key, &node); err != nil {
+		return nil, fmt.Errorf("error getting node %q: %v", node.Name, err)
+	}
+
+	glog.Infof("Machine %q is backing node %q", machine.Name, node.Name)
```

Comment:
nit: 'is backed by'

Reply:
isn't the machine backing the node, and the node backed by the machine?

Reply:
OK; I read it three ways. Happy as-is.